src/content_scripts/autocorrect.js

```diff
@@ -130,22 +136,13 @@ function insertIntoPage(atext) {
 /**
  * Count Unicode characters.
  * Adapted from: https://blog.jonnew.com/posts/poo-dot-length-equals-two
- * Intl.Segmenter is not yet supported by Firefox/Thunderbird: https://bugzilla.mozilla.org/show_bug.cgi?id=1423593
  *
  * @param {string} str
  * @returns {number}
  */
 function countChars(str) {
     // removing the joiners
-    const split = str.split("\u{200D}");
-    let count = 0;
-
-    for (const s of split) {
-        // removing the variation selectors
-        count += Array.from(s.replaceAll(/[\uFE00-\uFE0F]/gu, "")).length;
-    }
-
-    return count;
+    return Array.from(segmenter.segment(str.replaceAll("\u200D", ""))).length;
```

Comment:
```suggestion
    const STRING_JOINER_CHAR = "\u200D"
    return Array.from(segmenter.segment(str.replaceAll(STRING_JOINER_CHAR, ""))).length;
```

Another example on how you can remove the explanatory comment above and explain the code… with code itself.

Reply:
See my comment in Unicodify: https://github.com/rugk/unicodify/pull/96#discussion_r1891798336

Reply:
still the style could clarify things